ravenframework/Distributions1D.py

```diff
@@ -0,0 +1,854 @@
+    """
+    return self.dist.sf(x)
+
+  def untrQuantile(self, x):
+    """
+      Quantile function (inverse CDF) of the untruncated distribution.
+
+      @ In, x, float, quantile value to evaluate at
+      @ Out, ppf, float, quantile function value
+    """
+    return self.dist.ppf(x)
+
+  def untrMean(self):
+    """
+      Mean of the untruncated distribution.
+
+      @ In, None
+      @ Out, mean, float, mean of the untruncated distribution
+    """
+    return self.dist.mean()
+
+  def untrMedian(self):
+    """
+      Median of the untruncated distribution.
+
+      @ In, None
+      @ Out, median, float, median of the untruncated distribution
+    """
+    return self.dist.median()
+
+  def untrStdDev(self):
+    """
+      Standard deviation of the untruncated distribution.
+
+      @ In, None
+      @ Out, std, float, standard deviation of the untruncated distribution
+    """
+    return self.dist.std()
+
+  def untrMode(self):
+    """
+      Mode of the untruncated distribution.
+
+      @ In, None
+      @ Out, mode, float, mode of the untruncated distribution
+    """
+    # scipy.stats distributions don't have a mode function. Implementing this is left to the subclasses.
+    raise NotImplementedError("Mode is not implemented for scipy distributions.")
+
+  def untrHazard(self, x):
+    """
+      Hazard function of the untruncated distribution.
+      The hazard function is the ratio of the PDF to the survival function.
+
+      @ In, None
+      @ Out, mean, float, mean of the untruncated distribution
+    """
+    return self.dist.pdf(x) / self.dist.sf(x)
+
+
+class BasicUniformDistribution(ContinuousDistribution):
+  """ Uniform distribution wrapper """
+  def __init__(self, lowerBound, upperBound, xMin=None, xMax=None):
+    """
+      Class constructor
+
+      @ In, lowerBound, float, lower bound of the uniform distribution
+      @ In, upperBound, float, upper bound of the uniform distribution
+      @ In, xMin, float, optional, lower bound of truncated distribution
+      @ In, xMax, float, optional, upper bound of truncated distribution
+      @ Out, None
+    """
+    super().__init__(scipy.stats.uniform(lowerBound, upperBound - lowerBound), xMin, xMax)
+
+
+class BasicNormalDistribution(ContinuousDistribution):
+  """ Normal distribution wrapper """
+  def __init__(self, mean, sd, xMin=None, xMax=None):
+    """
+      Class constructor
+
+      @ In, mean, float, mean
+      @ In, sd, float, standard deviation
+      @ In, xMin, float, optional, lower bound of truncated distribution
+      @ In, xMax, float, optional, upper bound of truncated distribution
+      @ Out, None
+    """
+    super().__init__(scipy.stats.norm(mean, sd), xMin, xMax)
+
+  def untrMode(self):
+    """
+      Mode of the untruncated distribution
+
+      @ In, None
+      @ Out, mode, float, mode of the distribution
+    """
+    # mode and mean are the same for the normal distribution
+    return self.untrMean()
+
+
```

Comment:
Most of the distributions in src/distributions/distribution_1D.cxx and crow/include/distributions/distribution_1D.h (except the ones used by distributionNDNormal.cxx such as BasicNormalDistribution) can be removed I think.

Ditto for the includes needed for them:
```
#include <boost/math/distributions/lognormal.hpp>
#include <boost/math/distributions/triangular.hpp>
#include <boost/math/distributions/exponential.hpp>
#include <boost/math/distributions/weibull.hpp>
#include <boost/math/distributions/gamma.hpp>
#include <boost/math/distributions/beta.hpp>
#include <boost/math/distributions/poisson.hpp>
#include <boost/math/distributions/binomial.hpp>
#include <boost/math/distributions/logistic.hpp>
#include <boost/math/distributions/bernoulli.hpp>
#include <boost/math/distributions/laplace.hpp>
#include <boost/math/distributions/geometric.hpp>
```

Reply:
As discussed in meeting, these changes should be in a separate pull request.